Partial View Single

rayolanderos opened the pull request:
-Created View single which now retrieves info from the Datastore (just name, id, and cover, for now) according to the GET id parameter.
If no id is provided or if a non-existant id is provided, the error.html is called with a variable "error_msg" which will change depending on page.

-added "page_name" and "logout_url" to view_all.

-added css to the layout, dynamic JS, and page detection for header styling.

rayolanderos committed:
Partial View Single

-Created View single which now retrieves info from the Datastore (just name, id, and cover, for now) according to the GET id parameter.
If no id is provided or if a non-existant id is provided, the error.html is called with a variable "error_msg" which will change depending on page.

-added "page_name" and "logout_url" to view_all.

-added css to the layout, dynamic JS, and page detection for header styling.

gleono commented on services/stream_service.py:
```diff
@@ -9,6 +9,8 @@ class CreateStream(webapp2.RequestHandler):
     def post(self):
 
         json_string = self.request.body
+        logging.info("json-string")
```
Is it still needed to log the json string?

rayolanderos replied:
Nope. That's my bad. Will remove.

gleono commented on www/layout.html:
```diff
@@ -9,22 +9,25 @@
     <link rel="icon" href="img/favicon.ico">
 
-    <title>Connexus - View All Streams</title>
+    <title>Connexus - {{ page_name.capitalize() }}</title>
```
Nice! Thanks.

gleono commented on www/layout.html:
```diff
@@ -39,24 +42,37 @@
         </button>
         <div class="collapse navbar-collapse" id="navbarResponsive">
           <ul class="navbar-nav ml-auto">
+            {% if page_name == 'manage': %}
```
I think we can save some duplicated lines here. The only difference the `<span>`. So consider writing the if-else inside the `<li>` something like:
```html
<li class="nav-item active">
             <a class="nav-link" href="/manage">Manage</a>
{% if page_name == 'manage': %}
             <span class="sr-only">(current)</span>
{% endif %}
</li>
```
This way we don't need the else statement ;)

rayolanderos replied:
This is a tricky one because it's the `<span>` and the "active" class. I might just remove the span since it's for crawlers and readers and makes no difference to our front end, but still, it's the class that makes it tricky. I didn't want to get overly fancy with a solution for time reasons, but if needed -- we can go back to this.

gleono replied:
Oh, so if it is only the class we can do `<li class="nav-item {%if page_name == 'manage' %}active{% endif %}">` 
Maybe a [filter](http://jinja.pocoo.org/docs/2.9/templates/#equalto) could help?

rayolanderos replied:
Yeah that's what i ended up doing (first suggestion, not the filter, but great article for other things!). I removed the span entirely and just left the active class as variable in a condensed line.. you can see it in the merge now ;)

gleono commented on www/layout.html:
```diff
@@ -87,7 +103,71 @@
     <!-- IE10 viewport hack for Surface/desktop Windows 8 bug -->
     <script src="vendor/bootstrap/js/ie10-viewport-bug-workaround.js"></script>
-
+    {% if page_name == 'manage' : %}
```
You could also move this code to the `manage.html` file and if you really want it to be organized, you can add another `{% block %}` for the scripts in this layout. That way we don't have to modify `layout.html` everytime we want to add more scripts to specific templates.

rayolanderos replied:
Cool! Thanks for the tip. Will look into it.

gleono reviewed (changes requested):
Looks great, please just review the comments. If you commit to this same branch, changes are immediately reflected.

rayolanderos committed:
Review updates to Pull 3

"stream_service.py" - Removed unnecessary logging
"layout.html" -  Removed reader span, condensed "active" detection for header

Pending:
Keeping variable scripts within templates vs layout.
